### PR TITLE
Add more debugging information for cardinality agg

### PR DESCRIPTION
**Pull request by nik9000**
This adds two extra bits of info to the profiler:
1. Count of the number of different types of collectors. This lets us figure
   out if we're using the optimization for segment ordinals. It adds a few
   more similar counters just for good measure.
2. Profiles the `getLeafCollector` and `postCollection` methods. These are
   non-trivial for some aggregations, like cardinality.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregator.java [hunk 1]
```diff
@@ -234,7 +258,7 @@ public static long memoryOverhead(long maxOrd) {
         private final SortedSetDocValues values;
         private final int maxOrd;
         private final HyperLogLogPlusPlus counts;
-        private ObjectArray<FixedBitSet> visitedOrds;
+        private ObjectArray<FixedBitSet> visitedOrds;  // Danger! This is not tracked by BigArrays!
```
Comment (polyfractal) [inline server/src/main/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregator.java:261]: ++

**Review by polyfractal — APPROVED** [review]
LGTM, I like this.  More info == less tea-leaf-reading! :)

**Review by iverase — COMMENTED** [review]
+1 to add more debugging. It seems we need to handle BWC?

**Comment by nik9000** [comment]
> +1 to add more debugging. It seems we need to handle BWC?

Ah! BWC. How I love you.

**Comment by cjcenizal** [comment]
Thanks for the ping @nik9000 ! I created https://github.com/elastic/kibana/issues/77372 to track work for surfacing this information in the UI.

**Comment by nik9000** [comment]
> I created [elastic/kibana#77372](https://github.com/elastic/kibana/issues/77372) to track work for surfacing this information in the UI.

Thanks!

**Pull request merged**